### PR TITLE
Implement RCC Playbook pages

### PR DESCRIPTION
**Pull request by jhonatan-lopes**
# Description

This work implements the models required for the Responsible Computing Challenge Playbook.

Tests and clean-up will be addressed in subsequent PRs, in order to not make this larger than it already is.

Link to sample test page: Create a `RCC Landing Page` and then its child pages. If using a deploy app or on local dev, factories have been created to auto-populate the db with content.
Related PRs/issues:
- #10424 
- #10425 
- #10426 
- #10427 
- #10441 
- #10428 
- #10431 
- #10432 
- #10433 
- #10434 
- #10435 
- #10436 
- #10466 
- #10438 
- #10439 

# Checklist

**Tests**
- ~~[ ] Is the code I'm adding covered by tests?~~ -> No, the tests will be added as a separate PR.

**Changes in Models:**
- [X] Did I update or add new fake data?
- [X] Did I squash my migration? -> I've kept four migrations (one for the Page models, one for the taxonomy models and two for the links between taxonomies and pages).
- [X] Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?

**Documentation:**
- ~~[ ] Is my code documented?~~
- ~~[ ] Did I update the READMEs or wagtail documentation?~~

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -231,6 +231,10 @@ ignore="H017"
 "network-api/networkapi/wagtailpages/templates/wagtailpages/tags/cta/petition.html" = "H008"
 "network-api/networkapi/wagtailpages/templates/wagtailpages/tags/homepage_image.html" = "H006,H013"
 "network-api/networkapi/wagtailpages/templates/wagtailpages/tags/picture_ratios.html" = "H006"
+"network-api/networkapi/templates/fragments/libraries/author_card.html" = "H006"
+"network-api/networkapi/templates/pages/rcc/detail_page.html" = "H006"
+"network-api/networkapi/templates/pages/rcc/library_page.html" = "H023"
```
Comment (tbrlpld) [inline pyproject.toml:236]: What do you think about fixing the lint issues instead of ignoring them?

**Review by tbrlpld — COMMENTED** [review]
@jhonatan-lopes Sorry, but this is a huge PR. Have you considered splitting this into smaller chunks?

**Comment by jhonatan-lopes** [comment]
Hi @tbrlpld, I'm sorry, I know it's a huge PR. The pages were quite intertwined with each other and the taxonomies, so that made separating this quite tricky. If you feel like there is a better way to do it, I'm open to suggestions.

File: pyproject.toml (hunk 1 above)
Comment (jhonatan-lopes) [reply pyproject.toml:236]: Will do that :+1:

**Comment by jhonatan-lopes** [comment]
I'll separate it into smaller chunks, then...

**Pull request closed without merging**